Initizialize open fisca italy

lorenzo-stacchio opened the pull request.

lorenzo-stacchio committed:
Initialize open-fisca-italy

- Translated variable names into Italian;
- Change references for italian names;
- Introduced new parameter "eta_lavorativa.yaml";
- All tests passed using the virtual environment of openfisca except the eta_pensionabile.yaml (which is a test we written) that use the new variable "isAgeRetirement" (defined in variables/demografia.py) written by us to try open-fisca Variables;
Co-authored-by: corrado-petrelli

lorenzo-stacchio committed:
Commit to fix test

- Now test "eta_pensionabile.yaml" pass with correct semantic value;
 - The variable "isAgeRetirement" in demografia.py is defined semantically and syntactically well.

fpagnoux commented on CHANGELOG.md:
```diff
@@ -0,0 +1,195 @@
+# Changelog
+
+## 2.1.0 - [#29](https://github.com/openfisca/country-template/pull/29) [#30](https://github.com/openfisca/country-template/pull/30)
```
This Changelog is the file that tells the history of the evolution of your model.

It is very useful when you start having re-users who don't directly work with you (for exemple, someone who develops a web app on top of your model). Following your Changelog, they can know what's new in your model, and decide to upgrade, or to wait.

I'd encourage you to edit this file to add some information about what you started doing: what part of the legislation you started modeling, what reforms did you code, etc. 

The current content of this file is an example of Changelog. You can keep it somewhere as a reference, but it is not really about _your_ model, and it may confuse your re-users :).

lorenzo-stacchio replied:
You're totally right, i didn't really know how to modify this file. We will try to emulate the openfisca-country-template and openfisca-france changelog file and adapt it to our code.

fpagnoux commented on openfisca_italy/entita.py:
```diff
@@ -0,0 +1,61 @@
+# This file defines the entities needed by our legislation.
+from openfisca_core.entities import build_entity
+
+Household = build_entity(
```
> If you want, you could rename `Household` to `Famiglia `.

> It is up to your preference if you prefer using English or Italian words. In France for instance, we use French names, such as `Famille`. We find it more understandable for many re-users that are used to the legal French terminology.

lorenzo-stacchio replied:
Yeah sorry, this name has escaped us

fpagnoux commented on setup.py:
```diff
@@ -0,0 +1,33 @@
+
+setup(
+    name='OpenFisca-Italy',
+    version='2.1.0',
```
It seems odd to start a new model at version `2.1.0` :)

`OpenFisca-Core` and `OpenFisca-France` follow [semantic versioning](https://semver.org/).

SemVer is a set of rules about versioning that helps re-users to know whether they can upgrade their dependency without fearing to break something. The two main rules are
- we bump the left-most number for and only for non-backward compatible changes.
- If the left-most number is a 0, then backward compatibility is not guaranteed between two versions.

For instance, there can be a non backward compatible change (breaking change) from `1.1.2` to `2.0`, from `0.1.0` to `0.1.1`, but not from `1.1.2` to `1.2`

At an early stage, you probably don't want to bother with backward compatibility: you don't have users yet. To indicate that your code doesn't guarantee  backward compatibility, I'd suggest starting with `0.1.0`.

fpagnoux commented on openfisca_italy/situation_examples/__init__.py:
```diff
@@ -0,0 +1,17 @@
+
+
+single = parse('single.json')
+couple = parse('couple.json')
```
I think the name is `coppie.json` in your repository.

fpagnoux commented on openfisca_italy/variables/statistiche.py:
```diff
@@ -0,0 +1,46 @@
+    definition_period = MONTH
+    label = "Somma dei benefici percepiti da una famiglia"
+    reference = "https://stats.gov.example/benefits"
+    def total_benefits(household, period, parameters):
```
This doesn't work because a `formula` but be named `formula`, or `formula_2016` if it only starts from 2016.
Otherwise OpenFisca doesn't know what this function is and just raises an error.

lorenzo-stacchio replied:
Indeed, without comment this part, an error was raised and we didn't know why, thank for the explication.

fpagnoux reviewed (commented):
I put a few comments directly on the code.

I also notices that you only did two commits, and that the text of the first one contains 5 bullets points.

I'd recommend frequent small commits over big ones. Basically, each time you feel like you have done something, reached a goal, I'd suggest to commit. 

Big commits are risky: sometimes you do something that works, and then you start doing something else, and you realise you're going in the wrong direction, or breaking what you previously did. If you commit often, this is not a problem as you can quickly come back to the last commit when everything worked. 

It also helps understanding what has changed in the code. If a bug appears, you can spot which commit caused the bug. If it's a small commit, you can quickly understand the bug cause. If it's a huge one, good luck!  

Note that all those comments are just recommandations : you are building this model, and you are free to decide how to do it :).

Stach96 committed:
Fix to previous commit

- Writing Changelog.md;
- Better test for `eta_pensionabile.yaml` and comparison with the real parameter `eta_pensionamento` in `isAgeRetirement`.

Stach96 committed:
Add "eta_lavorativa"

- Add paramenter`eta_lavorativa`. a variable `isAgeOfWork` and the test `eta_lavorativa.yaml`

Stach96 committed:
BUG COMMIT

- Add test `tests/eta/maggiore_eta.yaml`for parameter `parameters/eta/maggiore_eta.yaml` and add variable "isAdult" in "demografia.py" to test it, seems that all works properly;
- There is a problem in test `tests/eta/eta_lavorativa.yaml`, only the last part (in which we tested the last period of relative parameter `parameters/eta/eta_lavorativa.yaml`). The variable used is `isAgeOfWork` in `demografia.py` that use the parameter `parameters/eta_lavorativa.yaml`.
- All the other tests added seems to work properly.

lorenzo-stacchio committed:
Changes values

- Change all the `0` and `1` in tests into `False` and `True`

lorenzo-stacchio committed:
FIX COMMIT

-  Changed `age` formula to solve issues with dates. Now all tests seems to work correctly.
- Changed name of variable in `demografia.py` to using  `snake_case` instead of `camelCase`. 
- Changed tests that now pass.

lorenzo-stacchio committed:
Introduction of income

- Reorganization of directories spacename;
- Add variables relative to income, to calculate to calculate both annual and monthly gross income; 
- Test for the new variables to see if formulas are correct or not.
- Erase tests not adaptable to our variables;
- With this commit, all tests passed.

lorenzo-stacchio committed:
Modifying tests for income variables

-Adapt test to new income variables

lorenzo-stacchio committed:
Incomes not taxale

- Insert parameters that indicate the treshold within which incomes are not taxable (since 2017);
- Split land income into: manufactured income and landed income.

lorenzo-stacchio committed:
Introduction Oneri Deducibili

- Defined type of deductible charges and test to see if the calculation is right;
- Defined marginal tax rate scale for IRPEF aliquotes;
- Merge of the two test file `reddito_totale_lordo_mensile` and  `reddito_totale_lordo_annuale`in a unique test file called  `reddito_totale_lordo`.

lorenzo-stacchio committed:
Gross IRPEF

- Formula to calculate gross IRPEF and relative test.

lorenzo-stacchio committed:
Tax deductions and IRPEF

- Introduce tax deductions tipologies, calculation and relative tests;
- Complete the IRPEF formula using the tax deductions;
- In this version all tests passed.

Stach96 committed:
Income retirement threshold

- Defined income retiremen treshold;
- Reorganization of some variables.

lorenzo-stacchio committed:
Refactor of source of income

- Defined new sub-variables to make the calculation more correct;
- Adaptation of tests to new variables.

lorenzo-stacchio committed:
Refactor of source income 2

-  Defined new sub-variables to make the calculation more correct;
- Adaptation of tests to new variables.

lorenzo-stacchio committed:
Init deductions calculation

- Defined deduction for firm income (ACE facilitation);
- Adapt IRPEF formula to the new Deduction:
- Adapt IRPEF tests to changes.

lorenzo-stacchio committed:
Income from amateur sports activity

- Defined income from amateur sports activity;
- Adapt IRPEF formula to this new income tipology;
- Adapt tests for this new income topology;

lorenzo-stacchio committed:
Deduction for investment in startup

- Defined deduction for investment in startup;
- Adapt IRPEF formula to this new deduction tipology;
- Adapt tests for this new income topology;

lorenzo-stacchio committed:
Introduce income not taxable by IRPEF

- Introduce situation of  retirement income for what IRPEF is 0;
- Adapt IRPEF formula to this new income situation;
- Adapt test to this new income situation.

lorenzo-stacchio committed:
Introduce new incomes not taxable by IRPEF

- Introduce situation of  landed income for what IRPEF is 0;
- Adapt IRPEF formula to this new income situation;
- Adapt test to this new income situation.
- Add directory `deduction for familiar charges`

lorenzo-stacchio committed:
Bug fix

- Insert a return value in a variable that cause the fail of the tests.

lorenzo-stacchio committed:
Introduce deduction for a dependent spouse

- Defined of all formula depending on the income to calculate deduction for a dependet spouse;
- Defined what is a detraction income, that is used in te calculation of the deduction just defined.

lorenzo-stacchio committed:
Introduced last case for a dependent spouse

- Introduce variable  `la_persona_non_ha_un_coniuge_a_carico` and `reddito_coniuge_per_calcolo_detrazioni_coniuge_a_carico` to cover a case in the deduction for a depend spouse calculation.
- Introduced the paremeter `soglia_familiare_a_carico` to manage the case cited above.

lorenzo-stacchio committed:
Introduce deduction for dependent children

- Introduce variables for dependent children deduction calculation;
- Introduce parameters used in calculation;
- Introduce test for this deduction.

lorenzo-stacchio committed:
Introduce deduction for dependent work

- Introduce python variables file `detrazione_figli_a_carico.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in ` detrazioni_lavoro_dipedente.yaml`;
- Perfectioned some formulas of  deduction for dependent children.

lorenzo-stacchio committed:
Introduce deduction for retirement income

- Introduce python variables file `detrazione_redditi_di_pensione.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_redditi_di_pensione`;
- Perfectioned some formulas of  deductions.

lorenzo-stacchio committed:
End of deduction for retirmente income

- Introduce two last cases in variable file and test file.

lorenzo-stacchio committed:
Refactor variables structure

lorenzo-stacchio committed:
Introduce deduction for income assimilated to income dependent work 

- Introduce python variables file `detrazioni_per_assimilati_a_lavoro_dipendente_ed_altri_redditi.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_redditi_assimilati_a_lavoro_dip_e_altri_redditi`;
- Introduce a file for the next deduction calls 'detrazioni_assegni_ex_coniuge.py'

lorenzo-stacchio committed:
Introduce deduction for ex-spouse checks

- Introduce python variables file `detrazioni_assegni_ex_coniuge.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_assegni_ex_coniuge.yaml`;
- Perfectione some test for `deduction for income assimilated to income dependent work`.

lorenzo-stacchio committed:
Introduce deduction for tenants of dwellings used as main dwellings

- Introduce python variables file `detrazioni_inquilini_di_alloggi_adibiti_ad_abit_principale.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_inquilini_di_alloggi_adibiti_ad_abit_principale.yaml`;

lorenzo-stacchio committed:
Introduce deduction for tenants of dwellings used as main dwellings with conventional contracts

- Introduce python variables file `detrazioni_inquilini_di_alloggi_ad_ab_principale_con_contratti_a_regime_convenzionale.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_inquilini_di_alloggi_adibiti_ad_abit_principale_con_contratti_a_regime_convenzionale.yaml`;

lorenzo-stacchio committed:
Introduce deduction for rents relating to leases for a main residence for young people aged between 20 and 30

- Introduce python variables file `detrazione_canoni_loc_rel_contratti_di_abitazione_per_giovani_tra_20_e_30_anni.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazione_canoni_locazione_relativi_a_contratti_di_abitazione_per_giovani_tra_20_e_30_anni.yaml`;

lorenzo-stacchio committed:
Introduce deduction for rent due to employees who transfer their residence for work purposes

- Introduce python variables file `detrazioni_canoni_loc_per_lavoratori_dip_che_trasf_per_lavoro.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_canoni_locazione_per_lavoratori_dip_che_si_trasferiscono_per_lavoro.yaml`;

lorenzo-stacchio committed:
Introduce deduction for rent of agricultural land for young people

- Introduce python variables file `detrazioni_per_affitto_di_terreni_agricoli_ai_giovani.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `detrazioni_per_affitto_di_terreni_agricoli_ai_giovani.yaml`;

lorenzo-stacchio committed:
Total deduction for rents and leases

lorenzo-stacchio committed:
Introduction total family and work deduction

lorenzo-stacchio committed:
Introduce residual credit and deduction used to deduct rentals and rents

- Introduce python variables file `credito_residuo_da_detrazioni_locazione_affitto.py` tha contains all the variables and relative formulas to calculate the total deduction;
- Parameters used in the formulas;
- Test for deduction in `credito_residuo_e_detrazione_fruita.yaml`;

lorenzo-stacchio committed:
Introduce expenses for what there is a deduction of 19/26 %

- Introduce python variables file `spese_per_le_quali_spetta_la_detrazione_19_26.py` that contains all the variables and relative formulas to calculate the total deduction;
- Test for deduction in `spese_detrazione_19_26.yaml`;

lorenzo-stacchio committed:
Introduce deduction for deductions for deductible charges

- Introduce python variables file `oneri_detraibili_19_26.py` that contains all the variables and relative formulas to calculate the total deduction;
- Test for deduction in `oneri_detraibili_19_26.yaml`;

lorenzo-stacchio committed:
Introduce Expenses for building renovation and anti-seismic measures

- Introduce python variables file `spese_per_interventi_di_recupero_del_patr_edilizio.py` that contains all the variables for the expenses and `fonti_detr_spese_di_rec_patr_edil_misure_antisismiche.py` that contains all the variables for the deductions and relative formulas to calculate them;
- Test for deduction in `spese_per_interventi_di_recupero_del_patrimonio_edilizio_misure_antisismiche.yaml`;

lorenzo-stacchio committed:
Introduce deduction for expenses for furnishing buildings, renovated, young couples, VAT for home purchase

- Introduce python variables file `spese_arredo_immobili_ristrutturati_iva_per_acquisto_abitazione.py` that contains all the variables for the expenses and insert the formula to calculate this deduction in `fonti_detrazioni_imposta.py`
- Test for deduction in `spese_arredo_immobili_sezione_III_C_quadro_RP.yaml`;

lorenzo-stacchio committed:
Introduce deduction for Expenses for interventions aimed at energy saving

- Introduce python variables file `detrazioni_spese_final_risp_energetico.py` that contains all the variables for the expenses and insert the formula to calculate this deduction in `spese_per_interventi_per_risparmio_energetico.py`
- Test for deduction in `spese_per_interventi_finalizzati_a_risparmio_energetico.yaml`;

lorenzo-stacchio committed:
Introduce deduction for other deduction

- Introduce python variables file `altre_detrazioni.py` that contains all the variables for the expenses and insert the formula to calculate this deduction in `sezione_spese_altre_detrazione.py`
- Test for deduction in `altre_detrazioni_sezioneVI_quadro_RP.yaml`;

lorenzo-stacchio committed:
Introduce deduction for investments in startups

- Introduce python variables file `detr_inv_startup_decadenza_SU_rec_detr.py` that contains all the variables for the expenses and insert the formula to calculate this deduction in `investimenti_in_start_up_.py`
- Test for deduction in `investimenti_startup.yaml`;

lorenzo-stacchio committed:
Introduce other deductions and recovery deductions for startup decadence

- The new variable files are available on the folder `fonti_altre_detrazioni`.
- Test fot this new deductions are available on `altre_detrazioni_sezioneVI_quadro_RP`

lorenzo-stacchio committed:
Introduce deduction for residual tax deduction 2014

- Introduce the formulas to calculate this deduction in `residui_detrazione_imposta_2014.py` that use many variables in many files.
- Test for deduction in `residuo_detrazioni_imposta_2014.yaml`;

lorenzo-stacchio committed:
Introduce deduction for deductions for residual unused tax credits

- Introduce the formulas to calculate this deduction in `residui_detrazione_imposta_2015.py` that use many variables in many files.
- Introduce the formula to calculate the credits the credit for deductions for residual unused tax credits in `residui_detr_crediti_imposta_deduzioni.py`
- Test for deduction in `residuo_detrazioni_imposta_2015.yaml`;

lorenzo-stacchio committed:
Introduce deduction for residual tax deduction 2015/2016

- Introduce the formulas to calculate this deduction in `residui_detrazione_imposta_2016.py` that use many variables in many files.
- Introduce the formula to calculate the credits the credit for deductions for residual unused tax credits in `residui_detr_crediti_imposta_deduzioni.py`
- Test for deduction in `residuo_detrazioni_imposta_2016.yaml`;

lorenzo-stacchio committed:
Introduce deduction for investment in startup

- Introduce the formulas to calculate this deduction in `detr_inv_startup.py` that use many variables in many files.
- Introduce the formula to calculate the credits the credit for deductions for residual unused tax credits in `residui_detr_crediti_imposta_deduzioni.py`
- Test for deduction in `investimenti_startup.yaml`;

lorenzo-stacchio committed:
Introduce modification and calculation for total deduction

- Modified formula in `detrazioni_imposta`;
- Modified set_input_period for some variable used in the formula;

lorenzo-stacchio committed:
Update deductible charge #1

lorenzo-stacchio committed:
Update deductible charge #2

lorenzo-stacchio committed:
Update deductile charge #3

-Untested Commit

lorenzo-stacchio committed:
Update deductible charge #4

- Defined the last formulas and variables for calculation of deductible charge;
- Rename of birth in data_nascita;

lorenzo-stacchio committed:
Re-organization of directories #1

lorenzo-stacchio committed:
Introduce deduction for health costs for certain diseases

- Reorganization of directory `oneri_deducibili`
- Introduce variable in file `detrazioni_spese_sanitarie_per_determinate_patologie.py`

lorenzo-stacchio committed:
Renaming to fix variable length limits on windows

lorenzo-stacchio committed:
Introduce deductions for  health expenses for certain diseases

- Introduce the formulas to calculate this deduction in `detrazioni_spese_sanitarie_per_determinate_patologie.py` that use many variables in many files.
- Test for deduction in `detrazioni_spese_sanitarie_determinate_patologie.yaml`;
- Reorganization of test folders.

lorenzo-stacchio committed:
Introduced tax credit generating residues

- Introduce the variables and the formula to calculate this deduction in `crediti_imposta_generano_residui.py`.
- Test for deduction in `crediti_imposta.yaml`;
- This code just introduce the concept of `tax credits that generate residuals` that could be defined in a correct way once the tax frameworks CR and LM  are defined.

lorenzo-stacchio committed:
Update Irpef formula #1

- Update IRPEF formula in `IRPEF.py` and the relative tests in `IRPEF.yaml`.

lorenzo-stacchio committed:
Introduce formal folder structure #1

- Introduce new folder Quadro_Ra;
- Introduce Readme.md that describes this tax framework.

lorenzo-stacchio committed:
Introduce formal folder structure #2

- Introduce new folder Quadro_CR;
- Introduce Readme.md that describes this tax framework.

lorenzo-stacchio committed:
Introduce formal folder structure #3

- Introduce new folder Quadro_RB;
- Introduce Readme.md that describes this tax framework.

lorenzo-stacchio committed:
Introduce formal folder structure #3

- Introduce new folder Quadro_RC;
- Introduce Readme.md that describes this tax framework.

lorenzo-stacchio committed:
Introduce formal folder structure #4

- Introduce new folder Quadro_Frontespizio;
- Introduce Readme.md that describes this tax framework.
- Introduce model as pdf file.

lorenzo-stacchio committed:
Introduce formal folder structure #5

- Introduce explicative PDF in all the tax frameworks;
- Introduce new tree structure for `Quadro_RP\Sezione_I`.

lorenzo-stacchio committed:
Introduce description of Section I

- Introduce module of RP tax framework at `quadri_oneri.pdf`;
- Introduced description of Section I in the relative README file.

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #1

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #2

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #2

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #4

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #5

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #6

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #7 and Update Sezione V Quadro RP

- Optimized some variables and files that concerns the fifth section of RP tax framework.

lorenzo-stacchio committed:
Sorting files in the new folder hierarchy #8 and Update Sezione III A Quadro RP

- Optimized some variables and files that concerns the fifth section of RP tax framework.

lorenzo-stacchio committed:
Update section III C of tax framework RP

lorenzo-stacchio committed:
Update RN1-RN4-RN5-RN26 and Sorting files in the new folder hierarchy #9

lorenzo-stacchio committed:
Renaming parameters folders and Update Sezione II

lorenzo-stacchio committed:
Update Sezione II

- Introduce variables and parameters for Rigo RP27;
- Introduce variables and parameters for Rigo RP28;

lorenzo-stacchio committed:
Introduce Rigo RP29 and Update Rigos RP27-RP28

lorenzo-stacchio committed:
Introduce Rigo RP30- RP31 and update Rigos from RP27 to RP29

lorenzo-stacchio committed:
Introduce Tax Framework for RP31

- Defined some varibles to describe the tax framework to compile the column 3 of row RP31 in QuadroRP\Sezione_II

lorenzo-stacchio committed:
Introduce Sezione_II\Rigo R32

lorenzo-stacchio committed:
Introduce RigoRP33-34-39

- End of Quadro RP Sezione II.

lorenzo-stacchio committed:
Introduce Section III A and Variables for RP41

lorenzo-stacchio committed:
Cloned RP41 mechanism for RP42-RP49

- Not optimized because of a duplication of codes that can be avoid with an import.

lorenzo-stacchio committed:
Update calculation of variables of Rigo RP48

lorenzo-stacchio committed:
Introduce RP51-RP52-RP53

- No one tests file were written because these are only descriptive rows.

lorenzo-stacchio committed:
Review and refactor Sezione III C

lorenzo-stacchio committed:
Introduced mechanism for compilation of row RP61

lorenzo-stacchio committed:
Introduce row RP62-63-64

- Clone the mechanism of row RP61 and adapt for these rows.
- Update calculation for variables in RP65.

tananai committed:
Introduced immobile

lorenzo-stacchio committed:
Correct some calculation on RP48

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

corradopetrelli committed:
Introduced "immobile" entity

corradopetrelli committed:
Merge branch 'Initizialize_open-fisca-italy' of https://github.com/openfisca/openfisca-italy into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Renaming and update existing variables

lorenzo-stacchio committed:
Introduce tax Framework LC1

- Tax framework relative to Coupon on the leases

corradopetrelli committed:
Introduced immobile.py for IMU

TO REFACTOR!

corradopetrelli committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Introduce TaxFramework LC1

- Introduce rows LC1 and LC2 relative to the calculation of Coupon on the leases  and the determination of the deposit.

corradopetrelli committed:
Changes immobile.py and testing

I test some feature of Immobile.py and I change "definition_period" from YEAR to MONTH.
For example I can change my main house on July.

lorenzo-stacchio committed:
Introduce __init__.py file in directories

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

corradopetrelli committed:
Update immobile.py

I introduce all variable for IMU
Unfortunately I can't insert a constraint that allows of connect a "CategoriaCatastale" with a range of "aliquote"

corradopetrelli committed:
Merge branch 'Initizialize_open-fisca-italy' of https://github.com/openfisca/openfisca-italy into Initizialize_open-fisca-italy

corradopetrelli committed:
Moving and Renaming "Immobile.py"

"Immobile.py" --> "immobile.py"
/immobile.py --> /Imu/immobile.py

lorenzo-stacchio committed:
Update calculation RN4 and load Enum error example

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Correct some brackets on IMU formulas

lorenzo-stacchio committed:
Update RN5 and linked variables

lorenzo-stacchio committed:
Commit following "Make enums work" model

corradopetrelli committed:
Testing on percentuale di possesso and fix immobile

corradopetrelli committed:
Merge branch 'Initizialize_open-fisca-italy' of https://github.com/openfisca/openfisca-italy into Initizialize_open-fisca-italy

corradopetrelli committed:
Resolve conflits

lorenzo-stacchio committed:
Fix some errors

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Fix merge conflicts

lorenzo-stacchio committed:
Update Section II of tax framework RP

corradopetrelli committed:
Introduced new tests and fix errors

lorenzo-stacchio committed:
Update Section II

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Update Section III A

corradopetrelli committed:
Introduce new test and fix minor error

- Test aree fabbricabili
- Test seconde case
- Test canone concordato
- Test immobili esenti che si trovano in determinate categorie
- Fix errors of immobile.py

lorenzo-stacchio committed:
Fix of circular check situation part I

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Update section II calcution of rows

lorenzo-stacchio committed:
Update Section III C tax framework RP

lorenzo-stacchio committed:
Update Section IV tax framework RP

lorenzo-stacchio committed:
Update Section V tax framework RP

corradopetrelli committed:
Refactor and Adding references

lorenzo-stacchio committed:
Update Section VI tax framework RP and RN1-2-3-4-5 tax framework RN

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Fixed Imu import

lorenzo-stacchio committed:
Name Refactor

lorenzo-stacchio committed:
Introduce ItalianTaxBenefitSystem and wrote reform example

lorenzo-stacchio committed:
Introduce Scenarios for Italian tax and benefit system

- In this commit, scenarios simulations doesn't work for a method contrast problem.

lorenzo-stacchio committed:
Load scenarios in ItalyTaxBenfitSystem

lorenzo-stacchio committed:
Fixed simulation scenarios problem and made first reform

lorenzo-stacchio committed:
Adding __init__.py files and changing test positions

lorenzo-stacchio committed:
Renaming variables of RP1-RP13

lorenzo-stacchio committed:
Renaming RP14-RP15

lorenzo-stacchio committed:
Write two reforms for IRPEF scale and aliquots

corradopetrelli committed:
Introduced Pertinenze Testing and Refactor

I introduced Pertinenze (a partitcular group of variable that touch the formula of imposta_imu).
Then I testing and refactor immobile.py

lorenzo-stacchio committed:
Update demography

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Syntax correction and introduce reform with right syntax

corradopetrelli committed:
Move folder Imu to Imposte

corradopetrelli committed:
Revert "Move folder Imu to Imposte"

This reverts commit c6533914e87148f24628d940b558ed77856e7fd7.

corradopetrelli committed:
Move from variables/Imu to variables/Imposte/Imu

corradopetrelli committed:
Move from tests/variables_parameters/imu to tests/variables_parameters/imposta/imu

corradopetrelli committed:
Refactor import of imu_common

corradopetrelli committed:
Introduced new tests and bug fixs

lorenzo-stacchio committed:
Update test, introduce some parameters

lorenzo-stacchio committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

corradopetrelli committed:
Update Obiettivi

corradopetrelli committed:
Revert "Update Obiettivi"

This reverts commit 1e4aec515f2242e3d82743aff6bc2f5d93647f98.

corradopetrelli committed:
Update ReadMe.MD feature

lorenzo-stacchio committed:
Merge pull request #4 from openfisca/corrado-petrelli-read-me

Update ReadMe.MD feature

lorenzo-stacchio committed:
Test some info system instruction

lorenzo-stacchio committed:
Modify setup. py

lorenzo-stacchio committed:
Add full test irpef

lorenzo-stacchio committed:
insert missing fields in fancy indexing

lorenzo-stacchio committed:
Change filename

lorenzo-stacchio committed:
Changes filename to fix window length problem

lorenzo-stacchio committed:
Refactor reforms filenames

corradopetrelli committed:
Rename Sezione I - RP14

lorenzo-stacchio committed:
Refactor QuadroRL filenames

lorenzo-stacchio committed:
Refactor deduction in Quadro RN

- Refactor names of child deduction

corradopetrelli committed:
Rename Sezione I - All

corradopetrelli committed:
Rename Sezione II

corradopetrelli committed:
Rename Sezione III

lorenzo-stacchio committed:
Refactor remaining filenames in Quadro RN

corradopetrelli committed:
Rename Sezione IV

corradopetrelli committed:
Rename Sezione V

corradopetrelli committed:
Rename Sezione VI

corradopetrelli committed:
Rename all py files referenced at Rename Sezione ***

corradopetrelli committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

corradopetrelli committed:
Bug Fix: Forget Rename lim_imp_deduc_contr_vers_lav_prima_occ

corradopetrelli committed:
Bug fix: Forget limite_spesa_acquisto_costruzione_immobile_dato_locazione

corradopetrelli committed:
Bug fix: Forget RP58_limite_spesa_arredo_immobile

lorenzo-stacchio committed:
Change filenames of test reform

lorenzo-stacchio committed:
Change some yaml test filenames

corradopetrelli committed:
Refactor all variables

corradopetrelli committed:
Merge remote-tracking branch 'origin/Initizialize_open-fisca-italy' into Initizialize_open-fisca-italy

lorenzo-stacchio committed:
Refactor some long filenames

lorenzo-stacchio committed:
Renaming problematic module

lorenzo-stacchio committed:
Renaming problematic modules #2
